pkg/plugins/builtin/http/http.go

```diff
@@ -68,6 +58,7 @@ type parameter struct {
 type auth struct {
 	Basic  authBasic `json:"basic"`
 	Bearer string    `json:"bearer"`
+	MTLS   mTLS      `json:"mTLS"`
```

Comment:
This applies to the new field, bu also the existing one. I suggest that we use pointers, to simplify the checks later in `validateConfig`. It would be easier to ensure that many auth methods are provided since they are mutually exclusive, without having to check every of their fields individually. Nothing prevent from keepin the current checks that ensure that all parameters required for an auth method in particular are present.

Reply:
Ok with that, I will change this.

Reply:
I think we have snake_case everywhere, `mTLS` looks out of place